deployments/constructs/carbyne-stack.ts

```diff
@@ -0,0 +1,577 @@
+/*
+ * Copyright (c) 2023 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository https://github.com/carbynestack/carbynestack.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as cdktf from "cdktf";
+import * as helm from "@cdktf/provider-helm";
+import { Construct } from "constructs";
+import * as path from "path";
+
+export interface CarbyneStackConfig {
+  dependsOn: cdktf.ITerraformDependable[];
+  helmProvider: cdktf.TerraformProvider;
+  fqdn: string;
+  partnerFQDN: string;
+  isMaster: boolean;
+  masterHost: string;
+  masterPort?: string;
+  noSSLValidation: boolean;
+  idPostfix?: string;
+  macKey: string;
+  prime: string;
+  r: string;
+  rInv: string;
+  gfpMacKey: string;
+  gf2nMacKey: string;
+  gf2nBitLength: number;
+  gf2nStorageSize: number;
+}
+
+export class CarbyneStack extends Construct {
+  constructor(scope: Construct, name: string, config: CarbyneStackConfig) {
+    super(scope, name);
+
+    const dependables: cdktf.ITerraformDependable[] = config.dependsOn;
+
+    const postgresDBMSChart = new cdktf.TerraformAsset(
+      this,
+      "postgres-dbms-chart-path",
+      {
+        path: path.resolve(__dirname, "../charts/postgres-dbms"),
+        type: cdktf.AssetType.DIRECTORY,
+      },
+    );
+
+    const postgresDBMS = new helm.release.Release(
+      this,
+      `postgres-dbms${config.idPostfix}`,
+      {
+        dependsOn: [...dependables],
+        provider: config.helmProvider,
+        wait: true,
+        waitForJobs: true,
+        name: "cs-postgres",
+        chart: `./${postgresDBMSChart.path}`,
+        set: [
+          { name: "users.cs[0]", value: "superuser" },
+          { name: "users.cs[1]", value: "createdb" },
+          { name: "users.cs[2]", value: "login" },
+          { name: "databases.castor", value: "cs" },
+          { name: "databases.amphora", value: "cs" },
+        ],
+      },
+    );
+
+    dependables.push(postgresDBMS);
+
+    const minio = new helm.release.Release(this, `minio${config.idPostfix}`, {
+      dependsOn: [...dependables],
+      provider: config.helmProvider,
+      wait: true,
+      waitForJobs: true,
+      name: "cs-minio",
```

Comment:
The `cs` prefix was originally derived from the release name chosen by the user. Is the release name now fixed, or would the deployment be something similar to `cs-cs-minio`?

Reply:
You are correct - the release name is hardcoded to `cs-*`.  When it transpiles it into HCL, it would would not get the extra `cs-cs-*` and seen in the generated HCL `/deployments/cdktf.out/stacks/local-kind/cdk.tf.json`:

```json
      "cs-1_minio-1_D396ADBD": {
        "//": {
          "metadata": {
            "path": "local-kind/cs-1/minio-1",
            "uniqueId": "cs-1_minio-1_D396ADBD"
          }
        },
        "chart": "minio",
        "name": "cs-minio",
        "provider": "helm.provider-helm-1",
        ...
    }
```

I can make this `cs-` prefix configurable if required.

Reply:
>Could you please create an issue to track this.

https://github.com/carbynestack/carbynestack/issues/93

>Please also include in this to check whether the numbers, e.g., minio-1, should be there really. 

You are correct - the top level construct: `VCP` & `CarbyneStack` provide the uniqueness Terraform requires.  It has been removed.